components/Buttons/src/MDCFloatingButton.m

```diff
@@ -273,19 +341,168 @@ - (void)layoutSubviews {
   self.titleLabel.frame = UIEdgeInsetsInsetRect(self.titleLabel.frame, self.titleEdgeInsets);
 }
 
-#pragma mark - Subclassing
+#pragma mark - Property Setters/Getters
+
+- (void)setMode:(MDCFloatingButtonMode)mode {
+  BOOL needsShapeUpdate = _mode != mode;
+  _mode = mode;
+  if (needsShapeUpdate) {
+    [self updateShape:YES];
+  }
+}
+
+- (void)setMinimumSize:(CGSize)size
+              forShape:(MDCFloatingButtonShape)shape
+                inMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToMinimumSize = self.shapeToModeToMinimumSize[@(shape)];
+  if (!modeToMinimumSize) {
+    modeToMinimumSize = [@{} mutableCopy];
+    self.shapeToModeToMinimumSize[@(shape)] = modeToMinimumSize;
+  }
+  modeToMinimumSize[@(mode)] = [NSValue valueWithCGSize:size];
+  if (shape == _shape && mode == _mode) {
+    [self updateShape:YES];
+  }
+}
+
+- (CGSize)minimumSizeForMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToMinimumSize = self.shapeToModeToMinimumSize[@(_shape)];
+  if(!modeToMinimumSize) {
+    return CGSizeZero;
+  }
 
-- (UIEdgeInsets)defaultContentEdgeInsets {
-  return UIEdgeInsetsZero;
+  NSValue *sizeValue = modeToMinimumSize[@(mode)];
+  if (sizeValue) {
+    return [sizeValue CGSizeValue];
+  } else {
+    return CGSizeZero;
+  }
 }
 
-- (UIEdgeInsets)defaultHitAreaInsets {
-  switch (_shape) {
-    case MDCFloatingButtonShapeDefault:
-      return UIEdgeInsetsZero;
-    case MDCFloatingButtonShapeMini:
-      // Increase the touch target from (40, 40) to the minimum (48, 48)
-      return UIEdgeInsetsMake(-4, -4, -4, -4);
+- (BOOL)updateMinimumSize {
+  CGSize newSize = [self minimumSizeForMode:self.mode];
+  BOOL sizeChanged = (newSize.width > self.minimumSize.width)
+      || (newSize.height > self.minimumSize.height);
+  super.minimumSize = newSize;
+  if (sizeChanged) {
+    [self invalidateIntrinsicContentSize];
+  }
+  return sizeChanged;
+}
+
+- (void)setMaximumSize:(CGSize)size
+              forShape:(MDCFloatingButtonShape)shape
+                inMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToMaximumSize = self.shapeToModeToMaximumSize[@(shape)];
+  if (!modeToMaximumSize) {
+    modeToMaximumSize = [@{} mutableCopy];
+    self.shapeToModeToMaximumSize[@(shape)] = modeToMaximumSize;
+  }
+  modeToMaximumSize[@(mode)] = [NSValue valueWithCGSize:size];
+  if (shape == _shape && mode == self.mode) {
+    [self updateShape:YES];
+  }
+}
+
+- (CGSize)maximumSizeForMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToMaximumSize = self.shapeToModeToMaximumSize[@(_shape)];
+  if (!modeToMaximumSize) {
+    return CGSizeZero;
+  }
+
+  NSValue *sizeValue = modeToMaximumSize[@(mode)];
+  if (sizeValue) {
+    return [sizeValue CGSizeValue];
+  } else {
+    return CGSizeZero;
+  }
+}
+
+- (BOOL)updateMaximumSize {
+  CGSize newSize = [self maximumSizeForMode:self.mode];
+
+  BOOL sizeChanged = !CGSizeEqualToSize(newSize, CGSizeZero)
```

Comment:
Maybe add a comment explaining why we only consider the size to have changed when it's *smaller* in at least one way compared to the current maximumSize?

Reply:
Good catch. Since a lot of this was prototype code, I don't think I thought it all the way through. The method should return `YES` on any dimensional change.

Reply:
Done